src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -247,4 +258,33 @@ public boolean verify(String hostname, SSLSession session) {
     };
     HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
   }
+
+  private void flushMessageBuffer() {
+    int batchSize = config.getBatchSize();
+    String jsonBatchKey = config.getJsonBatchKey();
+    boolean useJsonBatchKey = !Strings.isNullOrEmpty(jsonBatchKey) && batchSize > 1;
+    boolean shouldWriteJsonAsArray = config.shouldWriteJsonAsArray();
+    switch (config.getMessageFormat()) {
+      case "JSON":
+        String messageBody;
+        if (!shouldWriteJsonAsArray) {
+          messageBody = String.join(config.getDelimiterForMessages(), messageBuffer);
+        } else if (useJsonBatchKey) {
+          messageBody = String.format("{\"%s\":[%s]}", jsonBatchKey, String.join(",", messageBuffer));
```

Comment:
Manually manipulating strings like this is error prone. I'm assuming that's why there is that validation check on the characters in the jsonBatchKey, because this doesn't escape the key or anything? Instead, use a json library like the one being used by the StructuredRecordStringConverter. I think something like:

```
private static final JsonStructuredRecordDatumWriter JSON_DATUM_WRITER = new JsonStructuredRecordDatumWriter();
...

private void flushMessages() {
  ...
    StringWriter strWriter = new StringWriter();
    try (JsonWriter writer = new JsonWriter(strWriter)) {
      writer.beginObject().name(jsonBatchKey).beginArray();
      Encoder encoder = new JsonEncoder(writer)
      for (StructuredRecord record : messageBuffer) {
        JSON_DATUM_WRITER.encode(record, encoder);
      }
      writer.endArray().endObject();
    }
    ...
```

Reply:
Due to old gson version in cdap-api, we don't have .jsonValue function.

In new approach i have wrapped the records in a new structure record before calling the .toJsonString